src/main/docs/guide/httpServer/customHttpStatusCode

```diff
@@ -0,0 +1,58 @@
+The Micronaut framework supports Custom HTTP Status Codes . This feature enables developers to define and use HTTP status codes beyond the standard ones, allowing integration with systems or APIs that rely on non-standard status codes.
+
+
+=== Custom HTTP Status Codes
+Micronaut now supports arbitrary HTTP status codes that are not part of the predefined `HttpStatus` enum. Developers can specify any valid HTTP status code within the range `100-599`.
+
+==== Usage
+- `HttpResponse.status(int statusCode)`:
+Allows setting a custom HTTP status code.
+- `HttpResponse.status(int statusCode, String reason)`:
+Allows setting a custom HTTP status code along with a reason phrase.
+
+==== Examples
+[source,java]
+----
+HttpResponse.status(450, "Blocked by ISP");
+HttpResponse.status(599, "Network Connect Timeout Error");
+----
```

Comment:
please move all the hard coded snippets into references to actual source code for Groovy, Java and Kotlin using the `snippet::` macro